adapters/Unity/UnityAdapter/GADMediationAdapterUnity.m

```diff
@@ -30,68 +30,68 @@ @implementation GADMediationAdapterUnity
 // Called on Admob->init
 + (void)setUpWithConfiguration:(GADMediationServerConfiguration *)configuration
              completionHandler:(GADMediationAdapterSetUpCompletionBlock)completionHandler {
-  NSMutableSet *gameIDs = [[NSMutableSet alloc] init];
-  for (GADMediationCredentials *cred in configuration.credentials) {
-    NSString *gameIDFromSettings = cred.settings[kGADMAdapterUnityGameID];
-    GADMAdapterUnityMutableSetAddObject(gameIDs, gameIDFromSettings);
-  }
-
-  if (!gameIDs.count) {
-    NSError *error = GADMAdapterUnityErrorWithCodeAndDescription(
-        GADMAdapterUnityErrorInvalidServerParameters,
-        @"UnityAds mediation configurations did not contain a valid game ID.");
-    completionHandler(error);
-    return;
-  }
-
-  NSString *gameID = [gameIDs anyObject];
-  if (gameIDs.count > 1) {
-    NSLog(@"Found the following game IDs: %@. "
-          @"Please remove any game IDs you are not using from the AdMob UI.",
-          gameIDs);
-    NSLog(@"Initializing Unity Ads SDK with the game ID %@.", gameID);
-  }
-
-  [[GADMAdapterUnitySingleton sharedInstance] initializeWithGameID:gameID];
-  completionHandler(nil);
+    NSMutableSet *gameIDs = [[NSMutableSet alloc] init];
+    for (GADMediationCredentials *cred in configuration.credentials) {
+        NSString *gameIDFromSettings = cred.settings[kGADMAdapterUnityGameID];
+        GADMAdapterUnityMutableSetAddObject(gameIDs, gameIDFromSettings);
+    }
+    
+    if (!gameIDs.count) {
+        NSError *error = GADMAdapterUnityErrorWithCodeAndDescription(
+                                                                     GADMAdapterUnityErrorInvalidServerParameters,
+                                                                     @"UnityAds mediation configurations did not contain a valid game ID.");
+        completionHandler(error);
+        return;
+    }
+    
+    NSString *gameID = [gameIDs anyObject];
+    if (gameIDs.count > 1) {
+        NSLog(@"Found the following game IDs: %@. "
+              @"Please remove any game IDs you are not using from the AdMob UI.",
+              gameIDs);
+        NSLog(@"Initializing Unity Ads SDK with the game ID %@.", gameID);
+    }
+    
+    [[GADMAdapterUnity alloc] initializeWithGameID:gameID];
```

Comment:
you would need to create a new class `GADMUnityInitializationListener` which would implement protocol for InitializationDelegate.  you would need to add `initWithCompletionHandler` to `GADMUnityInitializationListener` so that you can initialize it with `completionHandler` from this method.

New parameter would be added to `[[GADMAdapterUnity alloc] initializeWithGameID:gameID];`, which is going be a `GADMUnityInitializationListener`.

Final code should look like:

```
GADMUnityInitializationListener* initializationListener = [[GADMUnityInitializationListener alloc] initWithCompletionHandler:completionHandler];
`[[GADMAdapterUnity alloc] initializeWithGameID:gameID initializationListener:initializationListener];`
```


`GADMUnityInitializationListener` should invoke provided `completionHandler`, so AdMob will be notified about initialization result.

Reply:
May I know what do you mean by "invoke provided completionHandler" here? I passed the completionHandler reference to GADMUnityInitializationListener, how can I update it as per the delegate's response?